### PR TITLE
Added 'show snmp' parser for IOSXR

### PR DESCRIPTION
**Pull request by muhammad-rafi**
## Description
Adding new parsers `show snmp` and `show snmp host` for iosxr.

## Motivation and Context
These parsers were not available for iosxr and can be really helpful to audit the snmp configs across many iosxr devices. I also added support for both ipv4 and ipv6. 

## Impact (If any)
None 

## Checklist:
- [x] I have updated the changelog.
- [ ] I have updated the documentation (If applicable).
- [x] I have added tests to cover my changes (If applicable).
- [x] All new and existing tests passed.
- [x] All new code passed compilation.

### CODE REVIEW

**Comment by tahigash** [comment]
@ThomasJRyan  and @lsheikal please review.

File: src/genie/libs/parser/iosxr/show_snmp.py [hunk 1]
```diff
@@ -0,0 +1,547 @@
...
+                snmp_dict["snmp_logging"]["logging_hosts"].update({logging_host: {"udp_port": udp_port}})
+
+                continue
+
+            m = p20.match(line)
+            if m:
+                group = m.groupdict()
+                notification_type = group["notification_type"]
+
+                # snmp_dict["snmp_logging"]["logging_hosts"][logging_host].update({"host_statistics": {"notification_type": notification_type}})
```
Comment (ThomasJRyan) [inline src/genie/libs/parser/iosxr/show_snmp.py:364]: Remove this

File: src/genie/libs/parser/iosxr/show_snmp.py [hunk 2]
```diff
@@ -0,0 +1,547 @@
...
+
+                host_statistics_dict["notification_type"] = notification_type
+                continue
+
+            m = p21.match(line)
+            if m:
+                group = m.groupdict()
+                pkts_in_trap_queue = int(group["pkts_in_trap_queue"])
+
+                # host_statistics_dict["pkts_in_trap_queue"] = pkts_in_trap_queue
```
Comment (ThomasJRyan) [inline src/genie/libs/parser/iosxr/show_snmp.py:377]: Remove this

File: src/genie/libs/parser/iosxr/show_snmp.py [hunk 3]
```diff
@@ -0,0 +1,547 @@
...
+
+        snmp_dict = {}
+
+        for line in output.splitlines():
+            if line:
+                line = line.strip()
+            else:
+                continue
+
+            m = p0.match(line)
```
Comment (ThomasJRyan) [inline src/genie/libs/parser/iosxr/show_snmp.py:183]: Above each match, please comment an example line that will be matched, exactly like you've done with the compiled regex

File: src/genie/libs/parser/iosxr/show_snmp.py [hunk 4]
```diff
@@ -0,0 +1,547 @@
...
+                # host_statistics_dict["pkts_in_trap_queue"] = pkts_in_trap_queue
+                host_statistics_dict.update({"pkts_in_trap_queue": pkts_in_trap_queue})
+                continue
+
+            m = p21.match(line)
+            if m:
+                group = m.groupdict()
+                pkts_in_trap_queue = int(group["pkts_in_trap_queue"])
+
+                # host_statistics_dict["pkts_in_trap_queue"] = pkts_in_trap_queue
```
Comment (ThomasJRyan) [inline src/genie/libs/parser/iosxr/show_snmp.py:386]: Remove this

File: src/genie/libs/parser/iosxr/show_snmp.py [hunk 5]
```diff
@@ -0,0 +1,547 @@
...
+        p3 = re.compile(r"^user:\s(?P<user>\S+)\s+security\smodel:\s(?P<version>\S+)\s(?P<v3_sec_level>\S+)$")
+
+
+        for line in output.splitlines():
+            if line:
+                line = line.strip()
+            else:
+                continue
+
+            if p0.match(line):
```
Comment (ThomasJRyan) [inline src/genie/libs/parser/iosxr/show_snmp.py:515]: Add comment above

**Review by ThomasJRyan — APPROVED** [review]
Overall exceptionally well done. Just needs to fix up comments and you're good to go

File: src/genie/libs/parser/iosxr/show_snmp.py [hunk 6]
```diff
@@ -0,0 +1,547 @@
...
+            m = p25.match(line)
+            if m:
+                group = m.groupdict()
+                informs_sent = int(group["informs_sent"])
+
+                if not snmp_dict["snmp_logging"].get("inform_statistics"):
+                    snmp_dict["snmp_logging"].update({"inform_statistics": {} })
+
+                snmp_dict["snmp_logging"]["inform_statistics"].update({"informs_sent": informs_sent})
+
```
Comment (lsheikal) [inline src/genie/libs/parser/iosxr/show_snmp.py:425]: add continue statement wherever applicable.

**Review by lsheikal — CHANGES_REQUESTED** [review]

File: src/genie/libs/parser/iosxr/show_snmp.py (hunk 2 above)
Comment (muhammad-rafi) [reply src/genie/libs/parser/iosxr/show_snmp.py]: Removed

**Comment by muhammad-rafi** [comment]
> Overall exceptionally well done. Just needs to fix up comments and you're good to go

Thank you for reviewing this and I have made changes as you requested, also found a minor which fixed too.

File: src/genie/libs/parser/iosxr/show_snmp.py (hunk 6 above)
Comment (muhammad-rafi) [reply src/genie/libs/parser/iosxr/show_snmp.py:437]: good catch, added missing continue statements

File: src/genie/libs/parser/iosxr/show_snmp.py (hunk 3 above)
Comment (muhammad-rafi) [reply src/genie/libs/parser/iosxr/show_snmp.py:184]: changes made as requested

File: src/genie/libs/parser/iosxr/show_snmp.py (hunk 5 above)
Comment (muhammad-rafi) [reply src/genie/libs/parser/iosxr/show_snmp.py:558]: added comments

**Review by lsheikal — APPROVED** [review]
Thank you for your contribution. Approved!!!

**Comment by muhammad-rafi** [comment]
> Thank you for your contribution. Approved!!!

Hey I am wondering if there is any action waiting from my side before it can be merged? 

Thanks,
-Rafi

**Comment by lsheikal** [comment]
> > Thank you for your contribution. Approved!!!
> 
> Hey I am wondering if there is any action waiting from my side before it can be merged?
> 
> Thanks, -Rafi

merged!!!

**Pull request merged**